nix/offchain.nix

```diff
@@ -1,12 +1,365 @@
+                default = false;
+              };
+
+              extraConfig = lib.mkOption {
+                description = '' FIXME '';
+                type = types.attrsOf types.anything;
+                default = { };
+              };
+            };
+          };
+
+          project = types.submodule {
+            options = {
+              src = lib.mkOption {
+                description = ''
+                  Path to the project's source code.
+
+                  Added in: 2.0.
+                '';
+                type = types.path;
+              };
+
+              censoredSpagoCodes = lib.mkOption {
+                description = '' FIXME '';
+                default = [ ];
+                type = types.listOf types.str;
+              };
+
+              shell = lib.mkOption {
+                description = '' FIXME '';
+                type = shell;
+                default = { };
+              };
+
+              bundles = lib.mkOption {
+                description = '' FIXME '';
+                type = types.attrsOf bundle;
+                default = { };
+              };
+
+              plutip = lib.mkOption {
+                description = '' FIXME '';
+                type = types.nullOr testConfigs;
+                default = null;
+              };
+
+              tests = lib.mkOption {
+                description = '' FIXME '';
+                type = types.nullOr testConfigs;
+                default = null;
+              };
+
+              runtime = lib.mkOption {
+                description = '' FIXME '';
+                type = runtime;
+                default = { };
+              };
+
+              enableFormatCheck = lib.mkOption {
+                description = '' FIXME '';
+                type = types.bool;
+                default = false;
+              };
+
+              enableJsLintCheck = lib.mkOption {
+                description = '' FIXME '';
+                type = types.bool;
+                default = false;
+              };
+            };
+          };
+        in
+        {
+          options.offchain = lib.mkOption {
+            description = "Off-chain project declaration";
+            type = types.attrsOf project;
+          };
+        });
   };
   config = {
-    perSystem = { config, self', inputs', pkgs, lib, ... }:
-      { };
+    perSystem = { config, self', inputs', lib, system, ... }:
+      let
+        defaultCtlOverlays = [
+          self.inputs.cardano-transaction-lib.overlays.purescript
+          self.inputs.cardano-transaction-lib.overlays.runtime
+        ];
+
+        liqwid-nix = self.inputs.liqwid-nix.inputs;
+        pkgs = import liqwid-nix.nixpkgs-ctl {
+          inherit system;
+          overlays =
+            [
+              self.inputs.cardano-transaction-lib.overlays.purescript
+              self.inputs.cardano-transaction-lib.overlays.runtime
+              self.inputs.cardano-transaction-lib.overlays.ctl-server
```

Comment:
I think `ctl-server` is deprecated in favor of `runtime`. We don't need it.

Reply:
I thought having it would give the annoying 
`trace: Warning: 'ctl-server' has moved to 'overlays.ctl-server' and will be removed from 'overlays.runtime' soon` but yeah, I think it will be fine to keep it there.

Reply:
It seems like even not having it gives the annoying logs, unfortunately.